lib/filter-stream.js

```diff
@@ -176,7 +177,7 @@ function FilterStream( blockMap, options ) {
     var buffer = Buffer.concat( this._buffers )
     var offset = 0
 
-    while( offset < buffer.length ) {
+    while( buffer.length - offset >= this.blockSize ) {
       this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
       offset = offset + this.blockSize
```

Comment:
Doesn't matter. I find `+=` more readable, but that's probably just a personal preference.